### PR TITLE
Add "shallow" option

### LINKED ISSUE #3: Shallow cloning support
For my current use case I only need the most recent state of the repository without the history. It would be great if the library supported shallow cloning (i.e. `git clone --depth 1`).

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves #3

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -20,9 +20,15 @@ class ClonerSource {
         this.dest = path.resolve(options.dest, input.path || this.source.name);
     }
     clone (cb) {
-        spawn(
+      let args = ["clone", this.source.toString(this.options.urlType), this.dest];
+
+      if (this.options.shallow) {
+        args.push("--depth=1");
```
Comment (IonicaBizau) [inline lib/index.js]: Can you please add 4 space indentation? 😁

**Comment by IonicaBizau** [comment]
This is cool! ✨

File: lib/index.js (hunk 1 above)
Comment (Turbo87) [reply lib/index.js]: oh sorry, missed that somehow... 🙈

Comment (Turbo87) [reply lib/index.js]: should be fixed!

**Comment by IonicaBizau** [comment]
Thanks! Landed in #6. :ship:

**Pull request merged**